src/layer/LayersFactory.ts

```diff
@@ -160,15 +161,86 @@ export class LayersFactory {
         if (!SHLayerClass) {
           throw new Error(`Dataset ${dataset.id} is not defined in LayersFactory.LAYER_FROM_DATASET`);
         }
-        return new SHLayerClass(
-          LayersFactory.parseSHInstanceId(baseUrl),
-          layerId,
-          null,
-          null,
-          null,
-          title,
-          description,
-        );
+
+        switch (dataset.id) {
+          case DATASET_AWS_S1GRD_IW.id:
+            return new S1GRDIWAWSLayer(
+              LayersFactory.parseSHInstanceId(baseUrl),
+              layerId,
+              null,
+              null,
+              null,
+              title,
+              description,
+              Polarization.SV,
```

Comment:
We can't assume that polarization is SV but we can't get the data from the layer either (except if we strictly limit the `layerId` options so that it must contain the polarization). Do you have any other idea?

Reply:
If user is authenticated (`setAuthToken` was called), we might be able to get this information from `/layers/` endpoint. However this is only needed in case `getMap` was called with Processing API - WMS doesn't need it as it is specified on the layer anyway. I would suggest fetching it lazily, similar to what we do with `backscatterCoeff` and `orthorectify` parameters. Note that they are not used by Processing API (but should be).

Reply:
As discussed live, I have rearranged Processing API support a bit. Dataset specific layers can now update the Processing API payload as they see fit, with the parameters they fetch (lazily) from the `/layers/` endpoint.

I have added the Storybook for S-1, but it depends on the appropriate layer being available. Not really useful unless the user is willing to dig in and test different configurations. The layer name is also hardcoded to `S1GRDIWDV` (I don't think making it configurable would be much of a help to users). In other words, we need to find a better way to configure storybook fixtures.

Can you maybe take a look at the changes and comment on them?

Reply:
Changes look good and make sense.

I tried without `Polarization.SV` here in `makeLayers()` and it works, so I think we don't need the switch (can be as it was before this MR).

Reply:
Thanks, good point! Since we don't supply `evalscript` at this step either, there is no reason to supply polarization (WMS knows it, Processing needs to ask service about other things anyway). I have removed the special initialization of S1 layer now.

Reply:
Nice. Resolving.

Reply:
Nice. Resolving.